src/components/PostGrid.tsx

```diff
@@ -34,53 +38,89 @@ const PostGrid = ({ data, user }: PostGridProps): JSX.Element => {
   }
 
   return (
-    <div className="bg-offWhite rounded-xl pt-6 px-4 pb-2 font-roboto">
-      <div className="w-full flex justify-between items-center mb-3">
-        <div className="flex w-full">
-          {data?.contributions[0] &&
-            <Avatar
-              contributor={data.contributions[0]?.contributor}
-              lastPr={dayjs(data.contributions[0]?.last_merged_at).fromNow()}/>}
+    <div className="">
+      <div className='border-[1px] border-gray-100 rounded-[14px]'>
+        <div className='flex flex-col rounded-t-[14px] bg-gray-100 p-[30px]'>
+          <div className='w-full flex justify-end '>
+            <div
+              role="button"
+              tabIndex={0}
+              aria-pressed="false"
+              onClick={() => user_id ? handleVoteUpdateByRepo(votes, repo_id) : signIn({ provider: 'github' })}
+              onKeyDown={(e) => {
+                if (e.key === 'Enter') {
+                  return user_id ? handleVoteUpdateByRepo(votes, repo_id) : signIn({ provider: 'github' });
+                }
+              }}
+              className="flex justify-center items-center text-base space-x-1 bg-gray-50 py-[2px] px-[5px] rounded-[5px] cursor-pointer transition-all duration-200"
+            >
+              <img src="/UpvoteIcon.png" alt="" />
+              <p className="font-bold text-grey130 ">{votes}</p>
+            </div>
+          </div>
+          
+          <a href={getRepoLink(full_name)}>
+            <div className='flex break-words items-center min-h-[100px] gap-[15px]'>
+              <div className='w-[50px] rounded-[14px] overflow-hidden'>
+                <img className='w-full h-auto' src={`https://avatars.githubusercontent.com/u/${repo_id}`} alt="" />
+              </div>
+              <div className='w-full break-all '>
+                <h6 className='text-[18px] tablet:text[22px] ' >{full_name}</h6>
+              </div>
+            </div>
+          </a>
 
-          {data?.contributions[1] &&
-            <Avatar
-              contributor={data.contributions[1]?.contributor}
-              lastPr={dayjs(data.contributions[1]?.last_merged_at).fromNow()}/>}
+        </div>
+        <div className='p-[30px] min-h-[106px] break-all'>
+          <p className='font-semibold text-grey60 text-[12px]'>{description.substring(0,100)}</p>
+        </div>
+      </div>
+      
+      <div className='flex justify-between py-[15px] w-full'>
+        <div className="flex gap-[6px] ">
+            {data?.contributions[0] &&
+              <Avatar
```

Comment:
Wouldn't it be cleaner if you used a FOR ? That I see only the index is changing 🥇

Reply:
Hey @bdougie, @NsdHSO 's suggestion is to use a for loop to render the "Avatar" components 5 times instead of writing it 5 times manually.

Reply:
@bdougie A little slice of Pizza :-), @shamimbinnur  can you take a look [here](https://stackblitz.com/edit/demo-react-class-component-suymdx?file=index.js,Counter.js) ?

Reply:
Perfect. Yeah we should do that here.

Reply:
On Our application we could we use like this
 `{data?.contributions?.map((item: DbContribution, index) => {
          if(index <= 5){
           return (<Avatar
            contributor={item?.contributor}
            lastPr={dayjs(item?.last_merged_at).fromNow()}
          />)}
        })}`

Reply:
_5contributions.map (contribution => (
              <Avatar
                contributor={contribution.contributor}
                lastPr={dayjs(contribution.last_merged_at).fromNow()}
              />
            ))

Reply:
@NsdHSO   We can use the slice function like this that is even more neat and specific.

`const _5contributions = data?.contributions.slice(0,5) || []

_5contributions.map (contribution => (
   <Avatar
   contributor={contribution.contributor}
   lastPr={dayjs(contribution.last_merged_at).fromNow()}
   />
 ))
`